### ISSUE TITLE
Version.created_at does not reflect when the version was created

### ISSUE THREAD

**Issue by diesl** [paper-trail-gem/paper_trail#1377]
Check the following boxes:

- [x] This is not a usage question, this is a bug report
- [x] This bug can be reproduced with the script I provide below
- [x] This bug can be reproduced in the latest release of the `paper_trail` gem

I am not sure if you are considering this as a bug, but I think that the `created_at` date of a model should always reflect the creation date of its instance (here: Version) and not the `updated_at` date of the versioned object (here: User).

```ruby
# frozen_string_literal: true

# Use this template to report PaperTrail bugs.
# Please include only the minimum code necessary to reproduce your issue.
require "bundler/inline"

# STEP ONE: What versions are you using?
gemfile(true) do
  ruby "3.0.2"
  source "https://rubygems.org"
  gem "activerecord", "6.1.4.1"
  gem "minitest", "5.11.3"
  gem "paper_trail", "12.2.0", require: false
  gem "sqlite3", "1.3.13"
end

require "active_record"
require "minitest/autorun"
require "logger"

# Please use sqlite for your bug reports, if possible.
ActiveRecord::Base.establish_connection(adapter: "sqlite3", database: ":memory:")
ActiveRecord::Base.logger = nil
ActiveRecord::Schema.define do
  # STEP TWO: Define your tables here.
  create_table :users, force: true do |t|
    t.text :first_name, null: false
    t.timestamps null: false
  end

  create_table :versions do |t|
    t.string :item_type, null: false
    t.integer :item_id, null: false
    t.string :event, null: false
    t.string :whodunnit
    t.text :object, limit: 1_073_741_823
    t.text :object_changes, limit: 1_073_741_823
    t.datetime :created_at
  end
  add_index :versions, %i[item_type item_id]
end
ActiveRecord::Base.logger = Logger.new(STDOUT)
require "paper_trail"

# STEP FOUR: Define your AR models here.
class User < ActiveRecord::Base
  has_paper_trail
end

# STEP FIVE: Please write a test that demonstrates your issue.
class BugTest < ActiveSupport::TestCase
  def test_1
    u = User.create(first_name: "Jane")
    sleep(2.seconds)

    assert_difference(-> { u.versions.last.created_at.sec }, +2) {
      u.paper_trail.save_with_version
    }
  end
end

# STEP SIX: Run this script using `ruby my_bug_report.rb`
```

In this example, I would expect the `created_at` date of the version created by `save_with_version` to be not the `updated_at` date of the original User object.

Currently, I have no way to know at which point of time this new version was created.

**Comment by jaredbeck** [comment]
I think you've found an interesting edge-case. Normally, we'd be making a change to the versioned record, e.g.

```diff 
     assert_difference(-> { u.versions.last.created_at.sec }, +2) {
+      u.first_name = "John"
       u.paper_trail.save_with_version
     }
```

### Root cause

If no change is made to the `User`, then `save`-ing it will not changed its `updated_at` timestamp.

### Workaround

As a workaround, you could use `touch` instead of `save_with_version`.

### History

Copying eg. `User#updated_at` into `Version#created_at` is an old feature. It was added almost eight years ago, in `5ee45827`. The reason isn't documented, but I guess they wanted the timestamps to match exactly. @straydogstudio Noel, do you remember the reason?

### Solutions?

Maybe, if nothing changed in the versioned record, we can detect that fact somehow in `PaperTrail::Events::Update#data` and choose not to override `created_at`. PRs welcome.

**Comment by diesl** [comment]
> I think you've found an interesting edge-case. Normally, we'd be making a change to the versioned record

My use case here is to take something like a snapshot of the model. Using `touch` would also change the timestamp of the `User`, which I would like to avoid.

I thought this is also the intended use case, as stated in the docs:

> Or, in rare situations, you may want to save a new version even if the record has not changed.

My workaround now is to update the timestamp of the `Version` right after creating it with `save_with_version`, but I thought this should be discussed/documented in general.

**Comment by diesl** [comment]
Not stale

**Comment by jaredbeck** [comment]
What's the next step here? PRs welcome.

**Comment by diesl** [comment]
> What's the next step here?

Probably this was not meant for me, but I think it is not really clear what is regarded as the desired solution:
1. Remove the code that copies `Model#updated_at` to `Version#created_at`, because it is not known why this happens
2. Detect that nothing changed in the versioned record and choose not to override `Version#created_at`
3. Just apply 1) if the time difference of `Model#updated_at` to `Version#created_at` is smaller than e.g. `1s`.
4. ...

**Comment by diesl** [comment]
Still not stale

**Comment by jaredbeck** [comment]
> Remove the code that copies Model#updated_at to Version#created_at, because it is not known why this happens

I assume that the goal of synchronizing timestamps is so that the primary record and its `Version` can be correlated by future reporting queries. I've always thought this is an odd feature, given that said correlation can be more reliably and performantly achieved by use of the foreign key (`item_id`). 

So, I'd like to suggest that we add an option to disable this feature.

For the new option's name, I'll suggest `synchronize_version_creation_timestamp`. It would be true by default.

```ruby
has_paper_trail(synchronize_version_creation_timestamp: false)
```

I'm open to disabling this feature by default, in a future major release, after a reasonable deprecation period.

If this approach works for you, a PR would be welcome.

Thank you for your contributions to this discussion so far, and your patience.

**Comment by SebRollen** [comment]
@jaredbeck @diesl, not sure if you received a notification from my linked PR yesterday, but figured I might as well explicitly bump that I've opened a PR that I think will address the issue here, using @jaredbeck's suggested option name.

It's my first contribution to paper_trail, so would appreciate any and all feedback on it!

**Comment by diesl** [comment]
@SebRollen Thanks for your PR! I did not have time yet to give feedback for your contribution, but I saw it got merged already 👍

**Comment by jaredbeck** [comment]
Fix released in 15.0.0
